### PR TITLE
[BIOX-1946] Copy Paste Support

### PR DESCRIPTION
**Pull request by VDykyi**
Implemented copy/paste and shape parameters bar

### CODE REVIEW

File: src/VAC/Global.cpp [hunk 1]
```diff
@@ -962,6 +965,36 @@ void Global::setPasteDelta(double delta)
     pasteDeltaY_ = delta;
 }
 
+const QRectF& Global::selectedGeometry() const
+{
+    return selectedGeometry_;
+}
```
Comment (marcus-sonestedt) [inline src/VAC/Global.cpp]: this and the other getter can be inline as the data is stored in the object without pimpl

File: src/VAC/VectorAnimationComplex/VAC.h [hunk 2]
```diff
@@ -483,6 +489,8 @@ public slots:
 
     double pasteDeltaX_;
     double pasteDeltaY_;
+
+    bool isManualTransform;
```
Comment (marcus-sonestedt) [inline src/VAC/VectorAnimationComplex/VAC.h]: add _ suffix to match other member var names?

File: src/VAC/VectorAnimationComplex/VAC.h [hunk 3]
```diff
@@ -158,7 +158,7 @@ class Q_VPAINT_EXPORT VAC: public SceneObject
     // -- Drag and drop --
     void prepareDragAndDrop(double x0, double y0, Time time);
     void performDragAndDrop(double x, double y);
-    void completeDragAndDrop();
+    void completeDragAndDrop(bool isEmitCheckpoint = true);
```
Comment (marcus-sonestedt) [inline src/VAC/VectorAnimationComplex/VAC.h]: "is" prefix is used only on getter functions, since we ask "is enabled?". There it is a command, so just name it "emitCheckpoint"

File: src/VAC/VectorAnimationComplex/TransformTool.cpp [hunk 4]
```diff
@@ -1099,6 +1104,77 @@ void TransformTool::endDragAndDrop()
...
+    for (auto cell : cells_)
+    {
+        obb.unite(cell->outlineBoundingBox(time));
+    }
+
+    if (obb.isProper())
+    {
+        double delta = (newHeight - obb.height()) / 2;
+
+        if (abs(delta) > EPS)
```
Comment (marcus-sonestedt) [inline src/VAC/VectorAnimationComplex/TransformTool.cpp]: I think Qt has some float equals function that handle EPS, prefer that if possible.

**Review by marcus-sonestedt — COMMENTED** [review]
Just nitpicks. Lots of changes, but all ok.

It's a bit hard to follow exactly how copy-paste works though (as I didn't code it). 
Maybe we should document it? Nothing big, but a bit more explanation would help us (and you :)  remember why we do things like we do 6 months from now.

**Review by marcus-sonestedt — APPROVED** [review]

File: src/VAC/Global.cpp (hunk 1 above)
Comment (VDykyi) [reply src/VAC/Global.cpp]: Done

File: src/VAC/VectorAnimationComplex/VAC.h (hunk 2 above)
Comment (VDykyi) [reply src/VAC/VectorAnimationComplex/VAC.h]: Done

File: src/VAC/VectorAnimationComplex/VAC.h (hunk 3 above)
Comment (VDykyi) [reply src/VAC/VectorAnimationComplex/VAC.h]: Done

File: src/VAC/VectorAnimationComplex/TransformTool.cpp (hunk 4 above)
Comment (VDykyi) [reply src/VAC/VectorAnimationComplex/TransformTool.cpp]: Done

**Comment by VDykyi** [comment]
> Just nitpicks. Lots of changes, but all ok.
> 
> It's a bit hard to follow exactly how copy-paste works though (as I didn't code it). Maybe we should document it? Nothing big, but a bit more explanation would help us (and you :) remember why we do things like we do 6 months from now.

I have added the several comments in code in last commit.
The copy/paste feature was implemented already in vpaint, I just expanded and improved its functionality in accoding to the our requirements.
